intel_extension_for_transformers/llm/runtime/graph/core/layers/inner_product.cpp

```diff
@@ -48,17 +48,22 @@ void JblasGemmCompF32(const int M, const int N, const int K, const float* A, con
                                              jblas::epilogue::gemm::CompFp32BlockEpilogue, custom::epilogue::AddFp32>;
     auto B = reinterpret_cast<typename Launcher::PrologueB::StorageWeight*>(_B);
     static Launcher kernel;
-    auto reduceA = kernel.mProA.createStorage(M, K, B->mBlockSize);
     utils::GemmProblem gp(1, M, N, K, B->mBlockSize);
+    auto reduceA = kernel.mProA.createReduceStorage(M, K, B->mBlockSize);
+    auto reordA = kernel.mProA.createReorderStorage(M, K, B->mBlockSize);
     if (B->IsAsym()) {
       reduceA.assign(WorkSpace);
+      WorkSpace += reduceA.mSize;
+    }
+    if (B->ShfIndice()) {
+      reordA.assign(WorkSpace);
```

Comment:
if no reduce and only shuffle, shall
```
WorkSpace += reordA.mSize;
```

Reply:
There is no storage left, so not necessary to shift the workspace pointer.